optuna/storages/_rdb/storage.py

```diff
@@ -584,11 +588,7 @@ def _update_trial(
         """
 
         with _create_scoped_session(self.scoped_session) as session:
-            trial_model = (
-                session.query(models.TrialModel)
-                .filter(models.TrialModel.trial_id == trial_id)
-                .one_or_none()
-            )
+            trial_model = models.TrialModel.find_or_raise_by_id(trial_id, session)
             if trial_model is None:
                 raise KeyError(models.NOT_FOUND_MSG)
```

Comment:
Looks like dead code that can be deleted, given your change above to use `find_or_raise_by_id`.
```suggestion
```